src/supa/connection/provider/server.py

```diff
@@ -170,9 +181,45 @@ def ReserveCommit(
         log = logger.bind(method="ReserveCommit")
         log.info("Received message.", request_message=pb_reserve_commit_request)
 
-        from supa import scheduler
+        connection_id = UUID(pb_reserve_commit_request.connection_id)
 
-        scheduler.add_job(ReserveCommitJob(UUID(pb_reserve_commit_request.connection_id)))
+        from supa.db.session import db_session
+
+        with db_session() as session:
+            reservation = (
+                session.query(model.Reservation).filter(model.Reservation.connection_id == connection_id).one_or_none()
+            )
+            try:
+                if reservation is None:
+                    raise NsiException(
+                        ReservationNonExistent, str(connection_id), {Variable.CONNECTION_ID: str(connection_id)}
+                    )
+                try:
```

Comment:
Not sure about the nested try and why that is needed

Reply:
I believe the original idea was to not repeat code, by nesting it this way there is only one place where you split between an NSI exception and any other exceptions, and the formatting of reason is left to NsiException.text(). If I flatten the code to remove one of the try/except and the raise NsiException's I will have to repeat sending the serviceException and formatting the reason. Neither way is perfect.

Reply:
Problem is by raising the NSIException you catch it again at line 206.

You could probably un nest the TransactionNotAllowed and it would perform the same way. It's not really necessary to catch it.  

If you want to remove duplication, its probably better to write an exception handling function.

Reply:
I simplified the code in both ReserveCommit and ReserveAbort so now the logic is more clear. For readability I added a skeleton send_service_exception() to supa.job.shared that will be implemented in a future commit.